File: fixtures/scale/python/large/cipherlab/checks/check_driftstream.py
from cipherlab.ciphers.driftstream import driftstream_encrypt, driftstream_decrypt
from cipherlab.specs.driftstream_spec import driftstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_driftstream():
    spec = driftstream_spec()
    for text in corpus_stream():
        sealed = driftstream_encrypt(text, spec["key"])
        opened = driftstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_driftstream.rb
require_relative 'driftstream_encrypt'
require_relative 'driftstream_decrypt'
require_relative 'corpus_stream'

def check_driftstream
  corpus_stream.each do |text|
    sealed = driftstream_encrypt(text, 9)
    opened = driftstream_decrypt(sealed, 9)
    return false if opened != text
  end
  true
end
